File: API/Models/Ad.cs
﻿using API.DAL;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using MongoDB.Driver.GeoJsonObjectModel;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace API.Models
{
    public class Ad:MongoEntity
    {
        [BsonRepresentation(BsonType.ObjectId)]
        public string categoryId { get; set; }

        public string title { get; set; }

        public string description { get; set; }
        public decimal price { get; set; }
        public IEnumerable<byte[]> pictures { get; set; }
        //public GeoJsonPoint<GeoJson2DGeographicCoordinates> location { get; set; }
        public Location location { get; set; }
        public Usuario postedBy { get; set; }
        public DateTime uploadDate { get; set; }
     
    }
}

File: Clients/Vendelo_RD_android/app/src/main/java/com/practices/samuel/vendelo_rd_android/Model/Ad.kt
package com.practices.samuel.vendelo_rd_android.Model

import java.util.*

/**
 * Created by Samuel on 12/23/2016.
 */
class Ad() : MongoEntity() {
    var categoryId:String = ""
    var title:String = ""
    var description:String=""
    var price:Double=0.0
    var pictures:List<ByteArray> =  emptyList<ByteArray>()
    var location:Location=Location(0.0,0.0)
    var usuario: Usuario? = null
    var uploadDate:Date?=null
}